File: iteration2/Models/Mandatory.java
package iteration2.Models;

import org.json.JSONObject;

import java.util.ArrayList;

public class Mandatory extends Course {

    public Mandatory(String name, String code, int credit, int requiredCredits, int quota, int semester,
                     ArrayList<Course> preRequisiteCourses, ArrayList<String> weeklyHours) {
        this.setName(name);
        this.setCode(code);
        this.setCredit(credit);
        this.setRequiredCredits(requiredCredits);
        this.setQuota(quota);
        this.setSemester(semester);
        this.setPreRequisiteCourses(preRequisiteCourses);
        this.setWeeklyHours(weeklyHours);
    }

    @Override
    public JSONObject toJson() {
        return super.toJson();
    }
}

File: iteration3/Models/Mandatory.py
import json
from typing import List

from iteration3.Models.Course import Course


class Mandatory(Course):
    def __init__(self, name, code, credit, required_credits, quota, semester, pre_requisite_courses, weekly_hours):
        super().__init__(name, code, credit, required_credits, quota, semester, pre_requisite_courses, weekly_hours)


    def to_json(self) -> json:
        return super().to_json()
